Sources/ZcashLightClientKit/Rust/ZcashRustBackend.swift

```diff
@@ -24,9 +24,43 @@ enum RustLogging: String {
     case trace
 }
 
+/// A description of the policy that is used to determine what notes are available for spending,
+/// based upon the number of confirmations (the number of blocks in the chain since and including
+/// the block in which a note was produced.)
+///
+/// See [`ZIP 315`] for details including the definitions of "trusted" and "untrusted" notes.
+///
+/// # Note
+///
+/// `trusted` and `untrusted` are both meant to be non-zero values.
+/// `0` will be treated as a request for a default value.
+///
+/// [`ZIP 315`]: https://zips.z.cash/zip-0315
+public struct ConfirmationsPolicy {
+    /// NonZero, zero for default
+    let trusted: UInt32
+    /// NonZero, zero for default, zero must match `trusted`
```

Comment:
I don't know what "zero must match `trusted`" means. At a guess it could mean that we must have `(trusted == 0) == (untrusted == 0)`. But why?

Reply:
```suggestion
    /// NonZero, zero for default; if this is set to zero, `trusted` must also be set to zero.
```
If I understand it correctly, the reason that we don't allow the defaults to be independent is that `untrusted` must be `>= trusted`? Since the caller doesn't necessarily know what the defaults are, they must choose either the defaults for everything or make their own explicit choices.

Reply:
There's a couple of levels of `ConfirmationPolicy` here. There's the Swift struct, then there's the FFI's struct and then there's Rust's struct. The Rust struct is the API we _want_, but forgive me if I'm wrong, we don't have a built-in way to express `NonZeroU32` in Swift (we could write it ourselves but that's beyond the scope of this PR, IMO).

If we expose any constructor of `ConfirmationsPolicy` that allows configuring `trusted` and `untrusted`, we have to interpret zeros as _something_, and so we're just doing the best we can with a less than ideal data modeling situation by using the default values in that case.

The code that actually does this checking is at the FFI-in-Rust level, [here](https://github.com/Electric-Coin-Company/zcash-light-client-ffi/pull/240/files#diff-8e458c9a55db6b2db3fed05c2fac6d6af36422148cf47b7c011219a8b05f82c9R1135-R1146).

In the latest set of changes I've moved the `default*` constructors to the `ConfirmationsPolicy` struct impl (in Swift) and changed the wording of the docs.